metric/metricexport/reader.go

```diff
@@ -172,6 +172,12 @@ func (ir *IntervalReader) Stop() {
 	ir.quit = nil
 }
 
+// Flush flushes the metrics by invoking ReadAndExport immediately.
+// The reader itself is not stopped and the timer is not reset.
+func (ir *IntervalReader) Flush() {
+	ir.reader.ReadAndExport(ir.exporter)
```

Comment:
do you know if this is safe to call concurrently?

Reply:
Thank you for reviewing this!

My bad. I have added some unit tests. They are also helpful in demonstrating the behavior of the code.
It seems that multiple concurrent calls will just export the metric multiple times. 
I'm not sure multiple calls to `Flush()` concurrent or not is the intended use case. I think this should be called once before shutdown. If it helps, I can make this into a `sync.Once` call.

Reply:
It sounds like the correct usage would always be to call `Flush` just after `Stop` to ensure they don't race, and to ensure metrics get sent:
```golang
ir.Start()
// do work
ir.Stop()
ir.Flush()
```
That way, you ensure you aren't sending anything else, and you empty the queue before shutting down (although I think it can still race and miss something?).  Normally, I would say we should combine Flush and Stop, but as the repo is in maintenance mode, I don't want to do anything that could cause regressions.

I think we should be able to use the quit and done channels to verify that it is being called after Stop(), or that the exporter has not been `Start'ed

Reply:
Made it only flush once after IR is stopped. Please let me know if this is acceptable. Thanks!

Reply:
I don't think we need to ensure flush is only called once--just that it isn't called with a started interval reader.  In theory, it should be possible to:
```golang
ir.Start()
ir.Stop()
ir.Flush()
ir.Start()
// ...
```

Reply:
That is a very good point, thank you!
I had to change the `Start()` function, I think it's a bug to check `ir.done` rather than `ir.quit` to see if IR is started.

Reply:
It looks like this still limits this to being called once.  Can we remove that bit?

Reply:
My bad, I misunderstood you. I thought you meant multiple calls with IR restart. I think this is fixed now. Thanks again.